### PR TITLE
Adjust HP penalties/bonuses based on beatmap specification

### PR DESCRIPTION
**Pull request by Artikash**
I think the formula is fine (gives hp 9-10 the same feeling of 99% 300s or die, hp0 of pretty much never die) but can be changed whenever.

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs [hunk 1]
```diff
@@ -23,11 +23,19 @@ public OsuScoreProcessor(RulesetContainer<OsuHitObject, OsuJudgement> rulesetCon
...
+        {
+            beatmapHp = beatmap.BeatmapInfo.Difficulty.DrainRate;
+        }
+
         protected override void Reset()
         {
             base.Reset();
 
-            Health.Value = 1;
+            Health.Value = 0.5;
```
Comment (MrTheMake) [inline osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: Why did this change happen?

Comment (Artikash) [reply osu.Game.Rulesets.Osu/Scoring/OsuScoreProcessor.cs]: To show me how much hp was gained while testing.

Oops.

**Comment by Artikash** [comment]
Undid the unintended hp starting at 50%, also refactored to pass the check.

**Review by peppy — DISMISSED** [review]

**Comment by peppy** [comment]
Obviously not perfect, but better than what we had.

Please make a new branch next time you conitrbute; don't use `master` as it is a pain to work with for others reviewing your fork.

**Review by peppy — APPROVED** [review]

**Pull request merged**